internal/component/unit.go

```diff
@@ -188,3 +216,125 @@ func (u *Unit) Dependents() Components {
 
 	return u.dependents
 }
+
+// String renders this unit as a human-readable string for debugging.
+func (u *Unit) String() string {
```

Comment:
same as with stacks: an example in the docstring would be helpful for readers and users